client/src/components/Rooms/Rooms.js

```diff
@@ -23,79 +23,98 @@ class Rooms extends Component {
-    // setTimeout(() => {
-    //   this.setState({
-    //     roomsDB,
-    //     FilteredRooms: roomsDB,
-    //     isLoading: false
-    //   });
-    // }, 1000);
     this.getDataFromDB(API.getRooms);
+    this.getDate();
   }
 
   componentDidUpdate(prevProps, prevState, snapshot) {
-    const { category, city } = this.state;
+    const { category } = this.state;
     if (prevState.category !== this.state.category) {
-      this.sortHandle(API.filterByCategory, category)
-    }
-    if (prevState.city !== this.state.city) {
-      this.sortHandle(API.filterByLocation, city)
+      this.filterHandle(API.filterByCategory, category)
     }
   }
 
   getDataFromDB = (api) => {
+    this.setState({isLoading: true});
     axios.get(api)
         .then(console.log('Data sorted by', api))
-        .then(res => this.setState({roomsDB: res.data.roomsData, isLoading: false}))
-        .catch(err => console.log(err))
+        .then(res => {
+          console.log('res===>', res)
+          this.setState({roomsDB: res.data, isLoading: false})
+        })
+        .catch(err => console.log(err));
   }
 
   postDataFromDB = (api, filter) => {
-    axios.post(api, filter)
+    this.setState({isLoading: true});
+    axios.post(api, {filter: filter})
         .then(console.log('Data filtered by', api))
-        // .then(res => this.setState({roomsDB: res.data.roomsData}))
-        .catch(err => console.log(err))
+        .then(res => {
+          console.log('res===>', res)
+          this.setState({roomsDB: res.data, isLoading: false})
+        })
+        .catch(err => console.log(err));
   }
 
   sortHandle = api => {
     this.getDataFromDB(api)
   }
 
   filterHandle = (api, filter) => {
-    this.postDataFromDB(api, filter)
+    filter ? this.postDataFromDB(api, filter) : this.getDataFromDB(API.getRooms);
   }
 
   sortRateHandle = () => {
-    const roomsAray = [...this.state.roomsDB];
-    roomsAray.sort((a, b) => b.rating - a.rating);
-    this.setState({ roomsDB: roomsAray });
+    this.sortHandle(API.sortRate)
   };
   sortMembersHandle = () => {
-    const roomsAray = [...this.state.roomsDB];
-    roomsAray.sort((a, b) => b.members - a.members);
-    this.setState({ roomsDB: roomsAray });
+    this.sortHandle(API.sortMember)
   };
   sortCreatedHandle = () => {
-    const roomsAray = [...this.state.roomsDB];
-    roomsAray.sort((a, b) => a.created_at - b.created_at);
-    this.setState({ roomsDB: roomsAray });
-    console.log("roomsAray", roomsAray);
+    this.sortHandle(API.sortCreated)
   };
   resetFiltersHandle = () => {
-    this.setState({ 
-      // roomsDB,
-      city: "",
+    this.getDataFromDB(API.getRooms);
+    this.setState({
       category: "",
     });
   };
+
   changeHandle = event => {
     this.setState({ [event.target.name]: event.target.value });
   };
 
+  getDate = () => {
+    const currentDate = new Date();
+    const date = this.formatDate(currentDate);
+    this.setState({ date });
+  };
+
+  formatDate = d => {
```

Comment:
I beleive we should use moment.js for handling dates. Maybe it is more reasonable add it and convert date via moment.js?